### PR TITLE
Enable security for prototyped APIs

### LINKED ISSUE #3118: Enable security for prototyped APIs
### Problem

Currently authentication is not required for APIs with prototyped lifecyclestatus within choreo connect. Since prototyped lifecycle state is changed to PreRelease, we need to enable authentication for those APIs. 

### Solution

Currently auth filter is not engaged for prototyped APIs. We need to change that logic such that auth filter is engaged.

### Implementation

_No response_

### Related Issues

_No response_

### Suggested Labels

_No response_

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose

$subject. 

Prototyped APIs should be validated using a valid token in order to access the API.

### Issues

Fixes #3118

### Automation tests
 - Unit tests added: No
 - Integration tests added: Yes

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Review by pubudu538 — APPROVED** [review]

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/PrototypedAPITestCase.java [hunk 1]
```diff
@@ -91,6 +91,6 @@ public void invokePrototypeAPISuccessTest() throws Exception {
                         Utils.getServiceURLHttps("/petstore-prototype/1.0.0/pet/findByStatus"), headers);
 
         Assert.assertNotNull(response);
-        Assert.assertEquals(response.getResponseCode(), HttpStatus.SC_OK,"Response code mismatched");
```
Comment (AmaliMatharaarachchi) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/PrototypedAPITestCase.java:94]: btw, don't we now need a success test as well?

**Review by AmaliMatharaarachchi — APPROVED** [review]

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/PrototypedAPITestCase.java (hunk 1 above)
Comment (VirajSalaka) [reply integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/PrototypedAPITestCase.java:94]: added via https://github.com/wso2/product-microgateway/pull/3117/commits/5065f20a52d7146c6639637f0ecdc71023311b34

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : dev-deployment-v2 : 20221024.11

**Comment by choreo-cicd** [comment]
[succeeded] DataplaneE1 cluster : dev-deployment-v2 : 20221024.11

**Comment by choreo-cicd** [comment]
[succeeded] DataplaneE2 cluster : dev-deployment-v2 : 20221024.11

**Comment by choreo-cicd** [comment]
[succeeded]  : dev-deployment-v2 : 20221024.12

**Comment by choreo-cicd** [comment]
[succeeded] DataplaneE1 cluster : stage-deployment-v2 : 20221026.1

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20221026.1

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : prod-deployment-v2 : 20221028.2

**Comment by choreo-cicd** [comment]
[failed] DataplaneE1 cluster : prod-deployment-v2 : 20221028.2

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20221028.2

**Pull request merged**